libs/blocks/footer/footer.js

```diff
@@ -1 +1,206 @@
+    navCols.forEach((column) => {
+      if (column) {
+        const navColumn = createTag('div', { class: 'footer-nav-column' });
+        const headings = column.querySelectorAll('h2');
+        headings.forEach((heading) => {
+          const navItem = createTag('div', { class: 'footer-nav-item' });
+          const titleId = heading.textContent.trim().toLowerCase().replace(/ /g, '-');
+          const title = createTag('a', {
+            class: 'footer-nav-item-title',
+            role: 'button',
+            'aria-expanded': this.desktop.matches,
+            'aria-controls': `${titleId}-menu`,
+          });
+          title.textContent = heading.textContent;
+          navItem.append(title);
+          const linksContainer = heading.nextElementSibling;
+          linksContainer.classList = 'footer-nav-item-links';
+          linksContainer.id = `${titleId}-menu`;
+          const links = linksContainer.querySelectorAll('li');
+          links.forEach((link) => {
+            link.classList.add('footer-nav-item-link');
+          });
+          navItem.append(linksContainer);
+          navColumn.append(navItem);
+        });
+        navGrid.append(navColumn);
+      }
+    });
+    if (navCols.length < 1) {
+      footer.classList.add('footer-small');
+    }
+    return navGrid;
+  };
+
+  decorateRegion = async () => {
+    const region = this.body.querySelector('.region-selector > div');
+    const regionTextContent = this.body.querySelector('.region-selector div > div')?.innerText;
+    if (!region) return null;
+    const regionContainer = createTag('div', { class: 'footer-region' });
+    const regionButton = createTag('a', {
+      class: 'footer-region-button',
+      id: 'region-button',
+      'aria-haspopup': true,
+      'aria-label': regionTextContent,
+      href: '#modal',
+      role: 'button',
+      tabindex: 0,
+    });
+    const regionText = createTag('span', { class: 'footer-region-text' }, regionTextContent);
+    regionButton.insertAdjacentHTML('afterbegin', GLOBE_IMG);
+    regionButton.append(regionText);
+    regionContainer.append(regionButton);
+
+    const regionLinks = region.querySelectorAll('a');
+    regionLinks.forEach((link) => {
+      const selected = link.parentNode.nodeName === 'STRONG';
+      const options = { class: 'footer-region-option' };
+      if (selected) {
+        options.class += ' footer-region-selected';
+        options['aria-current'] = 'page';
+      }
+    });
+    return regionContainer;
+  };
+
+  decorateSocial = () => {
+    const socialEl = this.body.querySelector('.social > div');
+    if (!socialEl) return null;
+    const socialWrapper = createTag('div', { class: 'footer-social' });
+    const socialLinks = createTag('ul', { class: 'footer-social-icons' });
+    socialEl.querySelectorAll('a').forEach((a) => {
+      const domain = a.host.replace(/www./, '').replace(/.com/, '');
+      const supported = ['facebook', 'instagram', 'twitter', 'linkedin'];
```

Comment:
nit: should this be a top level const? `SUPPORTED_SOCIAL` or something like that